File: 1Urte desberdinetako materiala/Lorea 25-26/Android/Erronka-1/multi_intefaz/src/main/java/model/Genre.java

package model;


public class Genre {
    private int liburua_id;
     private String generoa;

    public Genre(int liburua_id, String generoa) {
        this.liburua_id = liburua_id;
        this.generoa = generoa;
    }

    public Genre() {
    }

    public int getLiburua_id() {
        return liburua_id;
    }

    public void setLiburua_id(int liburua_id) {
        this.liburua_id = liburua_id;
    }

    public String getGeneroa() {
        return generoa;
    }

    public void setGeneroa(String generoa) {
        this.generoa = generoa;
    }

    @Override
    public String toString() {
        return "Genre{" + "liburua_id=" + liburua_id + ", generoa=" + generoa + '}';
    }

     
     
}

File: PMD 25-26/Erronka/app/src/main/java/com/example/erronka/model/Genre.kt
package com.example.erronka.model

/**class Genre {
    var liburua_id: Int = 0
    var generoa: String? = null

    constructor(liburua_id: Int, generoa: String?) {
        this.liburua_id = liburua_id
        this.generoa = generoa
    }

    constructor()

    override fun toString(): String {
        return "Genre{" + "liburua_id=" + liburua_id + ", generoa=" + generoa + '}'
    }
}**/
class Genre {

    private var liburua_id: Int = 0
    private var generoa: String = ""

    // Constructor con parámetros
    constructor(liburua_id: Int, generoa: String) {
        this.liburua_id = liburua_id
        this.generoa = generoa
    }

    // Constructor vacío
    constructor()

    // Getters y Setters
    fun getLiburua_id(): Int {
        return liburua_id
    }

    fun setLiburua_id(liburua_id: Int) {
        this.liburua_id = liburua_id
    }

    fun getGeneroa(): String {
        return generoa
    }

    fun setGeneroa(generoa: String) {
        this.generoa = generoa
    }

    // toString
    override fun toString(): String {
        return "Genre(liburua_id=$liburua_id, generoa=$generoa)"
    }
}